### PR TITLE
[NEW] Logout other user endpoint

### PR DESCRIPTION
**Pull request by ostjen**

### CODE REVIEW

**Review by pierre-lehnen-rc — CHANGES_REQUESTED** [review]
Rename the endpoint to just `users.logout` and the param to `userId`.
Make userId optional and when it's missing or the same as the currently logged user, then logout without checking the new permission.

File: app/api/server/v1/users.js [hunk 1]
```diff
@@ -933,3 +933,22 @@ API.v1.addRoute('users.listTeams', { authRequired: true }, {
 		});
 	},
 });
+
+API.v1.addRoute('users.logout', { authRequired: true }, {
+	post() {
+		const userId = this.bodyParams.userId || this.userId;
+
+		if (!hasPermission(this.userId, 'logout-other-user') && userId !== this.userId) {
```
Comment (pierre-lehnen-rc) [inline app/api/server/v1/users.js:941]:
```suggestion
		if (userId !== this.userId && !hasPermission(this.userId, 'logout-other-user')) {
```
If we compare the IDs first, then the hasPermission method will only be called when it is necessary.

File: tests/end-to-end/api/01-users.js [hunk 2]
```diff
@@ -3106,4 +3106,57 @@ describe('[Users]', function() {
...
+						.expect(200)
+						.end(done);
+				});
+		},
+		);
+
+		it('should logout the requester', (done) => {
+			updatePermission('logout-other-user', []).then(
+				() => {
+					console.log(user._id);
```
Comment (pierre-lehnen-rc) [inline tests/end-to-end/api/01-users.js:3152]:
```suggestion
```

Comment (ostjen) [reply tests/end-to-end/api/01-users.js:3152]: oops, mb

File: app/api/server/v1/users.js (hunk 1 above)
Comment (ostjen) [reply app/api/server/v1/users.js:941]: good call

**Review by pierre-lehnen-rc — APPROVED** [review]

**Comment by pbenkoe** [comment]
Will this endpoint log the (other) user out of all active sessions, on all devices?

Thank you!

**Comment by sampaiodiego** [comment]
yes it will @pbenkoe

**Pull request merged**